src/event.js

```diff
@@ -91,6 +91,36 @@ function on( elem, types, selector, data, fn, one ) {
 	} );
 }
 
+function addEventGetter( unused, name ) {
+	Object.defineProperty( jQuery.Event.prototype, name, {
+		enumerable: true,
+		configurable: true,
+
+		get: function() {
+			var value, hook;
+
+			if ( this.originalEvent ) {
+				if ( hook = jQuery.event.propHooks[ name ] ) {
+					value = hook( this.originalEvent );
+				} else {
+					value = this.originalEvent[ name ];
+				}
+			}
+
+			return this[ name ] = value;
+		},
+
+		set: function( value ) {
+			Object.defineProperty( this, name, {
```

Comment:
Setting may not be as rare as you think, especially in unit tests

```js
elem.trigger(jQuery.Event('click', { which: 1});
```

We would still need the setter, but both the setter and getter can just refer to the same key for storing the final value